### PR TITLE
refactor(zoe): separate zcf into zygotable methods

### PR DESCRIPTION
**Pull request by katelynsills**
This PR separates contractFacet.js into vatRoot.js and zcfZygote.js. vatRoot.js defines the zcf vat root and the methods that are exposed to the Zoe Service side. Currently, vatRoot.js is set up such that only one method on the root object is exposed, but when zygotes become a feature, we can expose more methods such that 1) creating a generic zcf, 2) evaluating contract code, and 3) inserting terms and other arguments, can be separately frozen.

Much of this code has only been moved. I will mark what has just been moved and what is new. 

Related to https://github.com/Agoric/agoric-sdk/issues/2860

### CODE REVIEW

File: packages/zoe/src/contractFacet/zcfZygote.js [hunk 1]
```diff
@@ -0,0 +1,353 @@
+// @ts-check
```
Comment (katelynsills) [inline packages/zoe/src/contractFacet/zcfZygote.js:1]: This file is nearly all code from contractFacet.js. I will mark the parts that are different.

File: packages/zoe/src/contractFacet/zcfZygote.js [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+export const makeZCFZygote = (
+  powers,
+  zoeService,
+  invitationIssuer,
+  testJigSetter,
+) => {
+  /** @type {PromiseRecord<ZoeInstanceAdmin>} */
+  const zoeInstanceAdminPromiseKit = makePromiseKit();
+  const zoeInstanceAdmin = zoeInstanceAdminPromiseKit.promise;
+
```
Comment (katelynsills) [inline packages/zoe/src/contractFacet/zcfZygote.js:37]: This code is new. In contractFacet.js, we had access to the zoeInstanceAdmin presence from the start. In this version, the zoeInstanceAdmin is not available until the last step, `zcfZygote.startContract`, so we will use a promise for a presence instead.

File: packages/zoe/src/contractFacet/zcfZygote.js [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+    },
+  });
+
+  let contractCode;
+
+  /** @type {ZCFZygote} */
+  const zcfZygote = {
+    evaluateContract: bundle => {
+      contractCode = evalContractBundle(bundle);
+      handlePWarning(contractCode);
```
Comment (katelynsills) [inline packages/zoe/src/contractFacet/zcfZygote.js:319]: This code is the same as before, just separated into the `evaluateContract` method.

File: packages/zoe/src/contractFacet/zcfZygote.js [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+    },
+    startContract: (
+      instanceAdminFromZoe,
+      instanceRecordFromZoe,
+      issuerStorageFromZoe,
+    ) => {
+      zoeInstanceAdminPromiseKit.resolve(instanceAdminFromZoe);
```
Comment (katelynsills) [inline packages/zoe/src/contractFacet/zcfZygote.js:326]: See the above comment for why we need to use a promise for the zoeInstanceAdmin

File: packages/zoe/src/contractFacet/zcfZygote.js [hunk 5]
```diff
@@ -0,0 +1,353 @@
...
+      instantiateInstanceRecordStorage(instanceRecordFromZoe);
+      instantiateIssuerStorage(issuerStorageFromZoe);
```
Comment (katelynsills) [inline packages/zoe/src/contractFacet/zcfZygote.js:328]: This code is the same as before, just moved to this last step.

File: packages/zoe/src/contractFacet/zcfZygote.js [hunk 6]
```diff
@@ -0,0 +1,353 @@
...
+            return harden({
+              creatorFacet,
+              publicFacet,
+              creatorInvitation,
+              handleOfferObj,
+            });
+          },
+        );
+      handlePWarning(result);
+      return result;
```
Comment (katelynsills) [inline packages/zoe/src/contractFacet/zcfZygote.js:349]: This code is the same as before, just moved into a method

**Review by erights — APPROVED** [review]
LGTM

**Pull request merged**